src/components/forms/EventStreamsForm.js

```diff
@@ -0,0 +1,200 @@
+import React, { Component } from "react";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import { IcseNameInput } from "../Inputs";
+import { EndpointSelect, IcseSelect } from "../Dropdowns";
+import { IcseFormGroup } from "../Utils";
+import { ToolTipWrapper } from "../Tooltips";
+import { isIpStringInvalid } from "../../lib/iam-utils";
+import { throughputAndStorageOptions } from "../../lib/es-utils";
+import { TextArea } from "@carbon/react";
+import { titleCase } from "lazy-z";
+import PropTypes from "prop-types";
+
+/**
+ * EventStreamsForm
+ */
+class EventStreamsForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = this.props.data;
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleAllowedIps = this.handleAllowedIps.bind(this);
+    this.handlePlanChange = this.handlePlanChange.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+  }
+
+  /**
+   * Handle input change for allowed ips text field
+   * @param {event} event
+   */
+  handleAllowedIps(event) {
+    // removing white space and checking for empty value
+    let value = event.target.value.replace(/\s*/g, "");
+    if (value === "") value = null;
+    this.setState({ private_ip_allowlist: value });
+  }
+
+  /**
+   * handle input change
+   * @param {event} event event
+   */
+  handleInputChange(event) {
+    this.setState(this.eventTargetToNameAndValue(event));
+  }
+
+  /**
+   * Handle input change for a select
+   * @param {event} event
+   */
+  handlePlanChange(event) {
+    let item = event.target.value;
+    this.setState({ plan: item.toLowerCase() });
+  }
```

Comment:
i think we should also reset values to defaults for everything that is not included in enterprise plan - right now, these fields will still be set on save because they are still persistent in state

<img width="345" alt="image" src="https://user-images.githubusercontent.com/51029454/228018080-71bea83a-e725-4619-a364-c3eb9254ee7e.png">